cmd/trade.go

```diff
@@ -506,7 +506,19 @@ func runTradeCmd(options inputs) {
 	botConfig = convertDeprecatedBotConfigValues(l, botConfig)
 	l.Infof("Trading %s:%s for %s:%s\n", botConfig.AssetCodeA, botConfig.IssuerA, botConfig.AssetCodeB, botConfig.IssuerB)
 
-	userID := "-1" // TODO DS Properly generate and save user ID.
+	var userIDPrehash string
+	if botConfig.IsTradingSdex() {
+		userIDPrehash = botConfig.TradingAccount()
+	} else {
+		userIDPrehash = amplitudeAPIKey
+	}
+
+	userIDHashed, e := utils.HashString(userIDPrehash)
+	if e != nil {
+		logger.Fatal(l, fmt.Errorf("could not create user id: %s", e))
+	}
+
+	userID := fmt.Sprint(userIDHashed)
```

Comment:
the newly added code is very self-contained. It may be cleaner if we put it in a separate function call, what do you think?
```
userID, e := getUserID(botConfig)
if e != nil {
    logger.Fatal(...)
}
```